### PR TITLE
Make it possible to configure maximum retry count

### PR DESCRIPTION
**Pull request by miretskiy**
Make it possible to configure maximum retry count when executing transactions.

### CODE REVIEW

File: crdb/tx.go [hunk 1]
```diff
@@ -101,6 +101,26 @@ func Execute(fn func() error) (err error) {
 	}
 }
 
+type txConfigKey struct {}
+
+// WithMaxRetries configures context so that ExecuteTx retries tx specified
+// number of times when encountering retryable errors.
+// Setting retries to 0 will retry indefinitely.
+func WithMaxRetries(ctx context.Context, retries int) context.Context {
```
Comment (rafiss) [inline crdb/tx.go:109]: could you add a basic test to verify something like

```
ctx = WithMaxRetries(ctx, n)
require.Equal(t, n, numRetriesFromContext(ctx)
```

**Review by rafiss — CHANGES_REQUESTED** [review]

File: crdb/tx.go (hunk 1 above)
Comment (miretskiy) [reply crdb/tx.go:109]: Done.

**Review by rafiss — APPROVED** [review]
thanks!

**Comment by rafiss** [comment]
i've tagged this in release v2.2.8

**Pull request merged**